sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/models/CreateDigitalTwinOptions.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.digitaltwins.core.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.util.Context;
+import com.azure.digitaltwins.core.implementation.models.DigitalTwinsAddOptions;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * The optional parameters for
+ * {@link com.azure.digitaltwins.core.DigitalTwinsClient#createDigitalTwinWithResponse(String, Object, Class, CreateDigitalTwinOptions, Context)} and
+ * {@link com.azure.digitaltwins.core.DigitalTwinsAsyncClient#createDigitalTwinWithResponse(String, Object, Class, CreateDigitalTwinOptions)}
+ */
+@Fluent
+public final class CreateDigitalTwinOptions {
+    /*
+     * Identifies the request in a distributed tracing system.
+     */
+    @JsonProperty(value = "traceparent")
+    private String traceparent;
+
+    /*
+     * Provides vendor-specific trace identification information and is a
+     * companion to traceparent.
+     */
+    @JsonProperty(value = "tracestate")
+    private String tracestate;
+
+    /**
+     * Get the traceparent property: Identifies the request in a distributed tracing system.
+     *
+     * @return the traceparent value.
+     */
+    public String getTraceparent() {
+        return this.traceparent;
+    }
+
+    /**
+     * Set the traceparent property: Identifies the request in a distributed tracing system.
+     *
+     * @param traceparent the traceparent value to set.
+     * @return the CreateDigitalTwinOptions object itself.
+     */
+    public CreateDigitalTwinOptions setTraceparent(String traceparent) {
+        this.traceparent = traceparent;
+        return this;
+    }
+
+    /**
+     * Get the tracestate property: Provides vendor-specific trace identification information and is a companion to
+     * traceparent.
+     *
+     * @return the tracestate value.
+     */
+    public String getTracestate() {
+        return this.tracestate;
+    }
+
+    /**
+     * Set the tracestate property: Provides vendor-specific trace identification information and is a companion to
+     * traceparent.
+     *
+     * @param tracestate the tracestate value to set.
+     * @return the CreateDigitalTwinOptions object itself.
+     */
+    public CreateDigitalTwinOptions setTracestate(String tracestate) {
+        this.tracestate = tracestate;
+        return this;
+    }
+
+    public DigitalTwinsAddOptions toProtocolLayerOptions() {
```

Comment:
I am not sure if we can have a public method that has a return type from the implementation package. 
Spoke to Tim offline and he is confirming that assumption.